app/full_text_search.py

```diff
@@ -42,15 +56,34 @@ def search(repo, query_txt, max_results):
     results = []
     if not query_txt:
         return results
-    index = appengine_search.Index(name=PERSON_FULL_TEXT_INDEX_NAME)
-    query_txt += ' AND (repo: ' + repo + ')'
+
+    query_txt = re.sub('|:|OR|AND|"', '', query_txt)
+
+    person_location_index = appengine_search.Index(
+        name=PERSON_LOCATION_FULL_TEXT_INDEX_NAME)
     options = appengine_search.QueryOptions(
         limit=max_results,
-        returned_fields=['record_id'])
-    index_results = index.search(appengine_search.Query(
-        query_string=query_txt, options=options))
-    for document in index_results:
-        id = document.fields[0].value
+        returned_fields=['record_id', 'names'])
+    and_query = query_txt + ' AND (repo: ' + repo + ')'
+    person_location_index_results = person_location_index.search(
+        appengine_search.Query(
+            query_string=and_query, options=options))
+
+    index_results = []
+    regexp = make_regexp(query_txt)
+    for document in person_location_index_results:
+        for field in document.fields:
+            if field.name == 'names':
+                names = field.value
+            else:
+                id = field.value
+
+        pattern = re.compile(regexp, re.I)
+        match = pattern.search(names)
+        if match:
+            index_results.append(id)
+
+    for id in index_results:
```

Comment:
You can simplify this part by:

    return [model.Person.get_by_key_name(repo + ':' + id) for id in index_results]